ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/MetricsCollector.scala

```diff
@@ -3,95 +3,82 @@
+        name: String,
+        value: MetricValue,
+        violatedObjective: Option[(ServiceLevelObjective[_], MetricValue)],
+    )
+    final case class FinalReport(totalDuration: Duration, metricsData: List[MetricFinalReportData])
+        extends Response
   }
 
   def apply[T](
-      streamName: String,
       metrics: List[Metric[T]],
-      logInterval: FiniteDuration,
-      reporter: MetricReporter,
       exposedMetrics: Option[ExposedMetrics[T]] = None,
-  ): Behavior[Message] =
-    Behaviors.withTimers { timers =>
-      val startTime: Instant = Instant.now()
-      new MetricsCollector[T](timers, streamName, logInterval, reporter, startTime, exposedMetrics)
-        .handlingMessages(metrics, startTime)
-    }
-
+  ): Behavior[Message] = {
+    val clock = Clock.systemUTC()
+    val startTime: Instant = clock.instant()
+    new MetricsCollector[T](exposedMetrics, clock).handlingMessages(metrics, startTime, startTime)
+  }
 }
 
-class MetricsCollector[T](
-    timers: TimerScheduler[MetricsCollector.Message],
-    streamName: String,
-    logInterval: FiniteDuration,
-    reporter: MetricReporter,
-    startTime: Instant,
-    exposedMetrics: Option[ExposedMetrics[T]],
-) {
+class MetricsCollector[T](exposedMetrics: Option[ExposedMetrics[T]], clock: Clock) {
   import MetricsCollector._
   import MetricsCollector.Message._
+  import MetricsCollector.Response._
 
-  timers.startTimerWithFixedDelay(PeriodicUpdateCommand(), logInterval)
-
-  def handlingMessages(metrics: List[Metric[T]], lastPeriodicCheck: Instant): Behavior[Message] = {
-    Behaviors.receive { case (context, message) =>
+  def handlingMessages(
+      metrics: List[Metric[T]],
+      lastPeriodicCheck: Instant,
+      startTime: Instant,
+  ): Behavior[Message] = {
+    Behaviors.receive { case (_, message) =>
       message match {
         case newValue: NewValue[T] =>
           exposedMetrics.foreach(_.onNext(newValue.value))
-          handlingMessages(metrics.map(_.onNext(newValue.value)), lastPeriodicCheck)
+          handlingMessages(metrics.map(_.onNext(newValue.value)), lastPeriodicCheck, startTime)
 
-        case _: PeriodicUpdateCommand =>
-          val currentTime = Instant.now()
+        case request: PeriodicReportRequest =>
+          val currentTime = clock.instant()
           val (newMetrics, values) = metrics
             .map(_.periodicValue(TimeUtil.durationBetween(lastPeriodicCheck, currentTime)))
             .unzip
-          context.log.info(namedMessage(reporter.formattedValues(values)))
-          handlingMessages(newMetrics, currentTime)
-
-        case message: StreamCompleted =>
-          context.log.info(
-            namedMessage(
-              reporter.finalReport(
-                streamName = streamName,
-                metrics = metrics,
-                duration = totalDuration,
+          request.replyTo ! Response.PeriodicReport(values)
+          handlingMessages(newMetrics, currentTime, startTime)
```

Comment:
When asked for a periodic report, compute one and respond via message (instead of logging it as it was done before).